src/java.desktop/windows/native/libawt/windows/awt_Component.cpp

```diff
@@ -6360,18 +6360,46 @@ void AwtComponent::_SetParent(void * param)
 {
     if (AwtToolkit::IsMainThread()) {
         JNIEnv *env = (JNIEnv *)JNU_GetEnv(jvm, JNI_VERSION_1_2);
-        SetParentStruct *data = (SetParentStruct*) param;
+        SetParentStruct *data = static_cast<SetParentStruct*>(param);
         jobject self = data->component;
         jobject parent = data->parentComp;
 
         AwtComponent *awtComponent = NULL;
         AwtComponent *awtParent = NULL;
 
-        PDATA pData;
-        JNI_CHECK_PEER_GOTO(self, ret);
-        awtComponent = (AwtComponent *)pData;
-        JNI_CHECK_PEER_GOTO(parent, ret);
-        awtParent = (AwtComponent *)pData;
+        if (self == NULL) {
```

Comment:
I had missed this in my earlier sweep of the changes, but didn't Phil request that both the check for self and parent be merged into self == NULL || parent == NULL? (Or as I'd prefer, nullptr)

Reply:
I noticed Phil's comment, but I could not make it look nice. For instance, the NPE error message was unclear, and you could not see the connection between self->awtComponent and parent->awtParent, which was obvious in the original code.

In the end, I thought it was preferable to use the exact same style when expanding the macro as elsewhere. That creates a common pattern across the codebase.